exporter/exporterhelper/internal/bounded_memory_queue_test.go

```diff
@@ -86,48 +74,27 @@ func TestBoundedQueue(t *testing.T) {
 func TestShutdownWhileNotEmpty(t *testing.T) {
 	q := NewBoundedMemoryQueue[string](1000)
 
-	consumerState := newConsumerState(t)
-
-	waitChan := make(chan struct{})
-	consumers := NewQueueConsumers(q, 5, func(_ context.Context, item string) error {
-		<-waitChan
-		consumerState.record(item)
-		return nil
-	})
-	assert.NoError(t, consumers.Start(context.Background(), componenttest.NewNopHost()))
+	assert.NoError(t, q.Start(context.Background(), componenttest.NewNopHost()))
+	for i := 0; i < 10; i++ {
+		assert.NoError(t, q.Offer(context.Background(), strconv.FormatInt(int64(i), 10)))
+	}
+	assert.NoError(t, q.Shutdown(context.Background()))
 
-	assert.NoError(t, q.Offer(context.Background(), "a"))
-	assert.NoError(t, q.Offer(context.Background(), "b"))
-	assert.NoError(t, q.Offer(context.Background(), "c"))
-	assert.NoError(t, q.Offer(context.Background(), "d"))
-	assert.NoError(t, q.Offer(context.Background(), "e"))
-	assert.NoError(t, q.Offer(context.Background(), "f"))
-	assert.NoError(t, q.Offer(context.Background(), "g"))
-	assert.NoError(t, q.Offer(context.Background(), "h"))
-	assert.NoError(t, q.Offer(context.Background(), "i"))
-	assert.NoError(t, q.Offer(context.Background(), "j"))
-
-	go func() {
-		assert.NoError(t, consumers.Shutdown(context.Background()))
-	}()
-
-	// wait a bit to ensure shutdown is called and unblock the consumers.
-	time.Sleep(100 * time.Millisecond)
-	close(waitChan)
-
-	consumerState.assertConsumed(map[string]bool{
-		"a": true,
-		"b": true,
-		"c": true,
-		"d": true,
-		"e": true,
-		"f": true,
-		"g": true,
-		"h": true,
-		"i": true,
-		"j": true,
-	})
+	assert.Equal(t, 10, q.Size())
+	numConsumed := 0
+	for i := 0; i < 10; i++ {
+		assert.True(t, q.Consume(func(ctx context.Context, item string) error {
+			assert.Equal(t, strconv.FormatInt(int64(i), 10), item)
+			numConsumed++
+			return nil
+		}))
+	}
+	assert.Equal(t, 10, numConsumed)
```

Comment:
We need to allow some time for the queue to be consumed? Shouldn't we use `assert.Eventually` to avoid flakiness?

Reply:
No, Consume is sync.